ServerCore/Pages/Teams/Create.cshtml.cs

```diff
@@ -89,6 +89,18 @@ public async Task<IActionResult> OnPostAsync()
                     return NotFound("Registration is full. No further teams may be created at the present time.");
                 }
 
+                //Isolate team name and Event ID
+                string teamName = Team.Name;
+                Event teamEvent = Team.Event;
+                int teamEventID = Team.Event.ID;
+                //Search for a team that matches Team Name and Event ID
+                Team duplicate = await _context.Teams.FirstOrDefaultAsync(t => t.Name == Team.Name && t.Event.ID == Team.Event.ID);
```

Comment:
Do you know if it's possible to do a case-insensitive comparison for team names? For example, we want to consider "TeAm NaMe" and "tEaM nAmE" as duplicates.

Reply:
Probable something like:
```
Team duplicate = await _context.Teams
    .FirstOrDefaultAsync(t => (t.Name.ToLower() == Team.Name.ToLower()) && 
                              (t.Event.ID == Team.Event.ID));
```

Reply:
If Teams is an actual .NET list then string.Compare is your best bet unless you have already restricted the team names to a safe set of characters. [See the docs](https://docs.microsoft.com/en-us/dotnet/api/system.string.compare?view=netframework-4.7.2#System_String_Compare_System_String_System_String_System_StringComparison_).

If that is some magic Linq-to-SQL query then I don't know.